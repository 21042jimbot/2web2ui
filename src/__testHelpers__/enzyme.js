export const findButton = (wrapper, buttonLabel) =>
  wrapper.findWhere(
    (node) => node.type() === 'button' && new RegExp(buttonLabel).test(node.text())
  );
