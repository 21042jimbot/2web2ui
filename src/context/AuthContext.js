import React, { createContext } from 'react';
import { connect } from 'react-redux';

export const AuthContext = createContext();

function Provider(props) {
  const { children, auth } = props;

  return <AuthContext.Provider value={auth}>{children}</AuthContext.Provider>;
}

function mapStateToProps(state) {
  return {
    auth: state.auth,
  };
}

export const AuthContextProvider = connect(mapStateToProps, {})(Provider);
