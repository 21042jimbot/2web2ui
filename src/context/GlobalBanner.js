import React, { createContext, Component } from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import moment from 'moment';

const defaultContext = {
  open: true
};

const ALWAYS_SHOW_ON_PATH = [
  '/account/billing',
  '/account/settings'
];

export const BannerContext = createContext(defaultContext);

export class GlobalBanner extends Component {

  render() {
    const { account, location, showBanner } = this.props;
    const { pending_cancellation } = account;

    const showBannerFromPath = !(
      (pending_cancellation && moment.duration(moment(pending_cancellation.effective_date).diff(moment.utc())).asDays() > 7) && //Checks if outside date
      !ALWAYS_SHOW_ON_PATH.includes(location.pathname) //Hide from the rest of the paths if outside dateRange
    );

    const value = {
      bannerOpen: Boolean(showBanner && //Banner state open
        pending_cancellation && showBannerFromPath)
    };

    return (
      <BannerContext.Provider value={value}>
        {/* Only components that are rendered under this tree will be able to "consume" */}
        {this.props.children}
      </BannerContext.Provider>
    );
  }
}

const mapStateToProps = ({ account, globalAlert }) => ({
  account,
  showBanner: globalAlert.showBanner
});
export default withRouter(connect(mapStateToProps)(GlobalBanner));
