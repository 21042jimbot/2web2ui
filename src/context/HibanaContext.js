import React, { createContext, useReducer, useContext } from 'react';

// See:
// https://kentcdodds.com/blog/how-to-use-react-context-effectively

const HibanaStateContext = createContext();
const HibanaDispatchContext = createContext();

const initialState = {
  isHibanaEnabled: false,
};

function useHibana() {
  return [useHibanaState(), useHibanaDispatch()];
}

function hibanaReducer(state, action) {
  switch (action.type) {
    case 'ENABLE': {
      return { isHibanaEnabled: true };
    }
    case 'DISABLE': {
      return { isHibanaEnabled: false };
    }
    default: {
      throw new Error(`Unhandled Hibana action type: ${action.type}`);
    }
  }
}

function HibanaProvider({ children }) {
  const [state, dispatch] = useReducer(hibanaReducer, initialState);

  return (
    <HibanaStateContext.Provider value={state}>
      <HibanaDispatchContext.Provider value={dispatch}>{children}</HibanaDispatchContext.Provider>
    </HibanaStateContext.Provider>
  );
}

function useHibanaState() {
  const context = useContext(HibanaStateContext);

  if (context === undefined) throw new Error('useHibanaState must be used within a HibanaProvider');

  return context;
}

function useHibanaDispatch() {
  const context = useContext(HibanaDispatchContext);

  if (context === undefined)
    throw new Error('useHibanaDispatch must be used within a HibanaProvider');

  return context;
}

export { HibanaProvider, useHibana };
