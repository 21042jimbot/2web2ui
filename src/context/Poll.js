import React, { createContext, Component } from 'react';
import moment from 'moment';

/**
 * PollContext
 * A simple polling service that persists between route changes
 * @prop startPolling func
 * @prop stoptPolling func
 *
 * @example
 * <PollContext.Consumer>
 *   {({ startPolling, stopPolling }) =>(
 *    <button onClick={() => startPolling(action)} />
 *    <button onClick={() => stopPolling(action.key)} />
 *   )}
 * </PollContext.Consumer>
 *
 * Action model:
 * {
 *   key: 'identifier', A string to represent this action
 *   interval: 1000, Time in ms between each poll
 *   duration: 5000, Time in ms to stop polling
 *   action: () => () The func to call at every interval until duration has reached
 * }
 */

const defaultContext = {
  actions: {}
};

const defaultAction = {
  interval: 1000,
  duration: 2000
};

export const PollContext = createContext(defaultContext);

/**
 * PollContext Provider
 */
class Poll extends Component {
  state = defaultContext;

  poll = (key) => {
    const { action, duration, startTime, interval, status } = this.state.actions[key];

    if (status === 'polling') {
      action();

      if (moment().diff(moment(startTime)) < duration) {
        setTimeout(() => this.poll(key), (interval));
      } else {
        this.setActionState(key, { status: 'done' });
      }
    }
  }

  start = (action) => {
    const { key, interval } = action;

    this.setActionState(key, {
      ...defaultAction,
      ...action,
      startTime: moment(),
      status: 'polling'
    });

    return setTimeout(() => this.poll(key), interval);
  };

  stop = (key) => {
    this.setActionState(key, { status: 'stopped' });
  };

  setActionState = (actionKey, values) => {
    this.setState({
      actions: {
        ...this.state.actions,
        [actionKey]: {
          ...this.state.actions[actionKey],
          ...values
        }
      }
    });
  }

  render() {
    const value = {
      actions: this.state.actions,
      startPolling: this.start,
      stopPolling: this.stop
    };

    return (
      <PollContext.Provider value={value}>
        {/* Only components that are rendered under this tree will be able to "consume" */}
        {this.props.children}
      </PollContext.Provider>
    );
  }
}

export default Poll;
