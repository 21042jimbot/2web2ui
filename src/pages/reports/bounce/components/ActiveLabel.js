import React from 'react';
import styles from './ActiveLabel.module.scss';

/**
 * Label that appears inside pie chart
 */
const ActiveLabel = ({
  name,
  value
}) => (
  <div className={styles.ActiveLabel}>
    <span className={styles.Label}>{name}</span>
    <span className={styles.Line}/>
    <span className={styles.Value}>{value}</span>
  </div>
);

export default ActiveLabel;
