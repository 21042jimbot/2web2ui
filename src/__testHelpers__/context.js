import React from 'react';
import { shallow } from 'enzyme';

/**
 * Shallow renders a component for React's context api
 * @example
 *   const consumer = context(<Button {...props}/>, { ...defaultContext });
 *   consumer.component.instance().doInstanceMethod();
 *   expect(consumer.children({ ...newContext })).toMatchSnapshot();
 */

export default function context(Component, context) {
  const component = shallow(Component);
  const ChildrenFunc = component.props().children;
  const children = newContext => shallow(<ChildrenFunc {...context} {...newContext} />);
  return { component, children };
}
