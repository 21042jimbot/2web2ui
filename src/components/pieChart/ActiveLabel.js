import React from 'react';
import PropTypes from 'prop-types';
import { Box, Text } from 'src/components/matchbox';
import useHibanaToggle from 'src/hooks/useHibanaToggle';
import styles from './ActiveLabel.module.scss';

/**
 * Label that appears inside pie chart
 */
function OGActiveLabel({ name, value }) {
  return (
    <div className={styles.ActiveLabel}>
      <div className={styles.Label}>{name}</div>
      <div className={styles.Line} />
      <div className={styles.Value}>{value}</div>
    </div>
  );
}

function HibanaActiveLabel({ name, value }) {
  return (
    <Box className={styles.ActiveLabel}>
      <Text as="div" mb="200" fontSize="400" color="gray.700">
        {name}
      </Text>

      <Text as="div" fontSize="400" color="gray.900" fontWeight="semibold">
        {value}
      </Text>
    </Box>
  );
}

export default function ActiveLabel(props) {
  return useHibanaToggle(OGActiveLabel, HibanaActiveLabel)(props);
}

ActiveLabel.propTypes = {
  name: PropTypes.string,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
};

ActiveLabel.displayName = 'PieChart.ActiveLabel';
