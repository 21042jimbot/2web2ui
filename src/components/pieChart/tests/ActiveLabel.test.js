import React from 'react';
import { shallow } from 'enzyme';
import { useHibana } from 'src/context/HibanaContext';
import ActiveLabel from '../ActiveLabel';

jest.mock('src/context/HibanaContext');

describe('ActiveLabel: ', () => {
  const props = {
    name: 'name',
    value: '123',
  };

  it('should render the OG version', () => {
    useHibana.mockImplementationOnce(() => [{ isHibanaEnabled: false }]);
    const wrapper = shallow(<ActiveLabel {...props} />);

    expect(wrapper.dive()).toMatchSnapshot();
  });

  it('should render the Hibana version', () => {
    useHibana.mockImplementationOnce(() => [{ isHibanaEnabled: true }]);
    const wrapper = shallow(<ActiveLabel {...props} />);

    expect(wrapper.dive()).toMatchSnapshot();
  });
});
