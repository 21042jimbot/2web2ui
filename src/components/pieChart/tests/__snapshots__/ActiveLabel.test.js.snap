// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActiveLabel:  should render the Hibana version 1`] = `
<Box
  className="ActiveLabel"
>
  <Text
    as="div"
    color="gray.700"
    fontSize="400"
    mb="200"
  >
    name
  </Text>
  <Text
    as="div"
    color="gray.900"
    fontSize="400"
    fontWeight="semibold"
  >
    123
  </Text>
</Box>
`;

exports[`ActiveLabel:  should render the OG version 1`] = `
<div
  className="ActiveLabel"
>
  <div
    className="Label"
  >
    name
  </div>
  <div
    className="Line"
  />
  <div
    className="Value"
  >
    123
  </div>
</div>
`;
